packages/webapp/src/pages/Activity.tsx

```diff
@@ -399,6 +399,40 @@ export default function Activity() {
         navigate(updatedUrl);
     }
 
+    const copyActivityLogUrl = (activity: ActivityResponse): string => {
+        let url = `${window.location.protocol}//${window.location.host}/activity?env=${env}`;
+        if (queryParams.activity_log_id) {
+            url = url.replace(`activity_log_id=${queryParams.activity_log_id}`, `activity_log_id=${activity.id}`);
+        } else {
+            url = url.concat(`&activity_log_id=${activity.id}`);
+        }
+
+        if (activity.connection_id) {
+            url = url.concat(`&connection=${activity.connection_id}`);
+        }
+
+        if (activity.provider_config_key) {
+            url = url.concat(`&integration=${activity.provider_config_key}`);
+        }
+
+        if (activity.operation_name) {
+            url = url.concat(`&script=${activity.operation_name}`);
+        }
+
+        if (activity.success) {
+            const status = activity.success === null ? 'in_progress' : activity.success ? 'success' : 'failure';
+            url = url.concat(`&status=${status}`);
+        }
+
+        if (activity.timestamp) {
+            const date = new Date(Number(activity.timestamp));
+            const dateString = `${date.getFullYear()}-${String(date.getMonth() + 1).padStart(2, '0')}-${String(date.getDate()).padStart(2, '0')}`;
+            url = url.concat(`&date=${dateString}`);
+        }
+
+        return url;
+    }
+
```

Comment:
Is it possible to use the URL interface instead of string maninpulation: ex:

```
const copyActivityLogUrl = (activity) => {

  const url = new URL(`${window.location.protocol}//${window.location.host}/activity`);

  url.searchParams.set('env', env);

  if (queryParams.activity_log_id) {
    url.searchParams.set('activity_log_id', activity.id); 
  } else {
    url.searchParams.append('activity_log_id', activity.id);
  }

  if (activity.connection_id) {
    url.searchParams.append('connection', activity.connection_id);
  }

...

  return url.toString();
}
```

Reply:
good suggestion, updated with https://github.com/NangoHQ/nango/pull/1567/commits/b2df8239e6df47360829221f29826b86cc299178